packages/auto-instrumentations-node/src/instrumentations/extentions/http.ts

```diff
@@ -91,14 +91,37 @@ function createHttpRequestHook(
       options,
       headers['content-encoding'] as string
     );
-    if (request instanceof IncomingMessage) {
+
+    if (request.constructor.name === 'ClientRequest') {
+      const clientRequest = request as ClientRequest;
```

Comment:
why needed?

Reply:
Because if the request is not of type ClientRequest then the line `request.write = function()...` won't work. It says: 'Property 'write' does not exist on type 'ClientRequest | IncomingMessage'